core/src/gateway_createfromjson.c

```diff
@@ -65,7 +67,7 @@ GATEWAY_HANDLE Gateway_CreateFromJson(const char* file_path)
                 {
                     properties->gateway_modules = NULL;
                     properties->gateway_links = NULL;
-                    if (parse_json_internal(properties, root_value) == PARSE_JSON_SUCCESS)
+                    if ((parse_json_internal(properties, root_value) == PARSE_JSON_SUCCESS) && properties->gateway_modules != NULL && properties->gateway_links != NULL)
```

Comment:
Yes. Before this logic was on the wrong place, it was on the JSON parser. By chaning this code I moved to the right place. Behavior stays the same as before.

---
In reply to: [108960877](https://github.com/Azure/azure-iot-gateway-sdk/pull/189#discussion_r108960877) [](ancestors = 108960877)